app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -335,7 +335,8 @@ def parse_config_pattern(config_pattern)
     def get_host_relationship(node)
       # Assign a physicalserver and host if server already exists and
       # some host match with physical Server's serial number
-      Host.find_by(:service_tag => node.serialNumber)
+      Host.find_by(:service_tag => node.serialNumber) ||
+        Host.joins(:hardware).where('hardwares.serial_number' => node.serialNumber).take
```

Comment:
Up to you how you want to handle it but my thought with using `.where` was we can check if `results.count > 1` and return nil so we don't have a false positive link.

Reply:
using find_by would automatically return a nil if the member was not found.   I will change this back to a find_by selector.  This will make rubocop happy as well.

Reply:
I was more worried about if there are duplicate BIOS UUIDs find_by would happily return the first one and we would have no idea that we might be linking to the wrong host.